include/osquery/config.h

```diff
@@ -69,7 +72,10 @@ class Config {
    *
    * @return a singleton instance of Config.
    */
-  static std::shared_ptr<Config> getInstance();
+  static Config& getInstance() {
```

Comment:
Also, you can avoid the "somewhat-weird" requirement to avoid the copy constructor at callsites by making the public methods (the Config API so to say) all static. Then within `load`, for example, any time you need to act on Config member data you prefix with `getInstance().` within the method, like: https://github.com/facebook/osquery/blob/master/osquery/registry/registry.cpp#L187-L199